### DISCUSSION TITLE
Why the SharpEngine behave differently on different GPUs？

### DISCUSSION THREAD

**Discussion by LZD121** [ab4d/Ab4d.SharpEngine.Samples#12] [Q&A]
I was involving SharpEngine in my Avalonia project, but I encountered a "divided by zero" error when I was running it on a computer with a NVIDIA RTX4070Ti Super GPU. However, it works fine on another computer which only has Intel UHD630 GPU.
Does anyone know why it behaves differently? How can I fix this error?

**Comment by abenedik** [comment]
There is a protocol in Avalonia ([[SPEC] External GPU memory interop (OpenGL, Vulkan, DirectX)](https://github.com/AvaloniaUI/Avalonia/issues/9925)) that is used by SharpEngine to ask what kind of shared texture is supported on the system. Based on that, a different code can be used to display the rendered 3D scene.

However, both NVIDIA and Intel GPU should use the same kind of shared texture, so I do not know why you got that error.

Could you please share a callstack from the error so that I can see where this happened (maybe this is in Avalonia or SkiaSharp's code). Does this happen with the SharpEngine samples project, or did you create your own project? In the later case, could you please share the source so that I can take a closer look (you can also share that by using the [Feedback page](https://www.ab4d.com/Feedback.aspx).
